src/custom/pages/Claim/Investment/InvestOption.tsx

```diff
@@ -0,0 +1,157 @@
+import { useCallback, useMemo } from 'react'
+import styled from 'styled-components/macro'
+import CowProtocolLogo from 'components/CowProtocolLogo'
+import { formatUnits, parseUnits } from '@ethersproject/units'
+import { CurrencyAmount } from '@uniswap/sdk-core'
+
+import { InvestTokenGroup, TokenLogo, InvestSummary, InvestInput } from '../styled'
+import { formatSmart } from 'utils/format'
+import Row from 'components/Row'
+import { CheckCircle } from 'react-feather'
+import { InvestOptionProps } from '.'
+import { ApprovalState } from 'hooks/useApproveCallback'
+import { useCurrencyBalance } from 'state/wallet/hooks'
+import { useActiveWeb3React } from 'hooks/web3'
+
+const RangeSteps = styled.div`
+  display: flex;
+  align-items: center;
+  width: 100%;
+  justify-content: space-between;
+`
+
+const RangeStep = styled.button`
+  background: none;
+  border: none;
+  font-size: 0.8rem;
+  cursor: pointer;
+  color: blue;
+  padding: 0;
+`
+
+export default function InvestOption({ approveState, approveCallback, updateInvestAmount, claim }: InvestOptionProps) {
+  const { currencyAmount, price, cost: maxCost, investedAmount } = claim
+
+  const { account } = useActiveWeb3React()
+
+  const token = currencyAmount?.currency
+
+  const balance = useCurrencyBalance(account || undefined, token)
+
+  const handlePercentChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    console.log(event.target.value)
+  }
+
+  const handleStepChange = (value: number) => {
+    console.log(value)
+  }
+
+  const onMaxClick = useCallback(() => {
+    if (!maxCost || !balance) {
+      return
+    }
+
+    const amount = maxCost.greaterThan(balance) ? balance : maxCost
+    // store the value as a string to prevent unnecessary re-renders
+    const investAmount = formatUnits(amount.quotient.toString(), balance.currency.decimals)
```

Comment:
why aren't you saving it as atoms? (may be @alfetopito who did this) but question remains

especially since you're using it in other places to do maths and need to reconvert. We usually just formatSmart the thing from atoms or am I wrong?